### PR TITLE
Build tools to examine routes in the repl

### PR DESCRIPTION
**Pull request by manandearth**
## Goal
Building test-tools for examining routes in the repl 
## PR status, roadblocks, etc
 I see the examples in pedestal http://pedestal.io/guides/developing-at-the-repl ... 
How to implement in `component` ?
First definition in the pedestal example is: `(def serv (run-dev))` that later is used to look into the running application in:
```
server> (require '[io.pedestal.test :refer [response-for]])
nil
server> (response-for (::server/service-fn serv) :get "/")
 ```

### CODE REVIEW

**Comment by manandearth** [comment]
What I'd like to be able to do is to test a route to see the response in the repl. The tests fail due to:
>  Caused by: java.net.BindException: Address already in use

File: src/playground/system_test.clj [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+(def url-for (route/url-for-routes
+              (route/expand-routes service/routes)))
+
+(defn service-fn
+  [system]
+  (get-in system [:pedestal :service ::http/service-fn]))
+
+(defmacro with-system
```
Comment (vemv) [inline src/playground/system_test.clj]: Delete this macro, it's not necessary given our setup

File: src/playground/system_test.clj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  [[bound-var binding-expr] & body]
+  `(let [~bound-var (component/start ~binding-expr)]
+     (try
+       ~@body
+       (finally
+         (component/stop ~bound-var)))))
+
+(deftest greeting-test
+  (with-system [sut (user/dev-system)]                       ;; <1>
+    (let [service               (service-fn sut)                 ;; <2>
```
Comment (vemv) [inline src/playground/system_test.clj]: Remove `with-system [sut (user/dev-system)]` entirely. You should never invoke (user/dev-system) by hand; that is Component's job.

Do this instead:

```
(let [system com.stuartsierra.component.repl/system ;; com.stuartsierra.component.repl/system holds your system after a `(reset)`
      service (service-fn system)]
  ...)
```

File: dev/user.clj [hunk 3]
```diff
@@ -15,9 +15,10 @@
    [clojure.string :as string]
    [com.grzm.component.pedestal :as pedestal-component]
    [com.stuartsierra.component :as component]
-   [com.stuartsierra.component.repl :refer [reset set-init]]
+   [com.stuartsierra.component.repl :refer [reset set-init system start stop]]
```
Comment (vemv) [inline dev/user.clj]: Generally, never use `start` or `stop`. Better remove them from here so you're not tempted.

`reset` is idempotent, particularly the 'rehardened' snippet which you should always use

**Comment by manandearth** [comment]
Got this to work, I need to start experimenting to see how the `comp-response-for` can further make the works of the interceptors clear.
What do I do with this `system-test` ns  and a helper function like `comp-response-for` ? move it to `/tests/playground/`?

File: dev/user.clj (hunk 3 above)
Comment (manandearth) [reply dev/user.clj]: I understood that from you before. :slightly_smiling_face:

File: src/playground/system_test.clj (hunk 2 above)
Comment (manandearth) [reply src/playground/system_test.clj]: thanks!

**Comment by manandearth** [comment]
I will merge this and move on to `buddy`, step by step to avoid the mess of the last branch. I will start with a simple register to db. no hashing in the first commit and hopefully by then I could get an indication of the way forward.

**Comment by vemv** [comment]
Nice work!

I'd say you should leave only _unit tests_ in `test/`. `dev/` is more apt for development facilities.

**Pull request merged**